### PR TITLE
#165273567  implemented JWT strategy for authentication

### PR DESCRIPTION
**Pull request by kalsmic**
#### What does this PR do?

- Implements JWT strategy for authentication.

#### Description of Task to be completed?

- Add JWT Token in login and Registration endpoint responses.
- Add tests for JWT

#### How should this be manually tested?

- Run tests with the  `pytest`  command in the terminal to run
 
 - In An API Consumer for example [Postman](https://www.getpostman.com/)

1. Register a user to verify if the JWT token in part of the response
2. Login into the application to verify if the JWT token is part of the response.
3. Try accessing the welcome route without an authorization header
4. Try accessing a protected route with an expired token.
5. Try accessing a protected route with a valid token

#### Any background context you want to provide?

-	The welcome route in the authentication may be removed later on when we add protected endpoints.

-	For now, the welcome endpoint  GET  `/api/welcome/` can be used for JWT tests for protected routes.

-	The Authorization header Is as follows `Bearer JWT_TOKEN`

-	The Token expires after 3 hours.
- Also, take note that the SECRET_KEY was added to the project environmental variables

#### What are the relevant pivotal tracker stories?

[#165273567](https://www.pivotaltracker.com/story/show/165273567 )

### CODE REVIEW

**Review by llwasampijja — DISMISSED** [review]
@kalsmic , I have noticed that the coverage has decreased by 1.05%, try to check and see if you missed out on testing any of the functions that you added.
Otherwise, good work!

File: authors/settings.py [hunk 1]
```diff
@@ -19,7 +19,7 @@
 # See https://docs.djangoproject.com/en/1.11/howto/deployment/checklist/
 
 # SECURITY WARNING: keep the secret key used in production secret!
-SECRET_KEY = '7pgozr2jn7zs_o%i8id6=rddie!*0f0qy3$oy$(8231i^4*@u3'
+SECRET_KEY = os.environ.get("SECRET_KEY")
```
Comment (JamesMudidi) [inline authors/settings.py:22]: Nice work here @kalsmic

**Review by JamesMudidi — APPROVED** [review]
Other than the coverage dropping, the work is good

**Review by bisonlou — APPROVED** [review]
Approved. This is good

**Review by JamesMudidi — APPROVED** [review]
Nice work

**Review by nadralia — APPROVED** [review]
Nice Work @kalsmic i like the idea of hiding the secret key

**Review by llwasampijja — APPROVED** [review]
Great that the coverage has increased!

**Review by ZackAtama — APPROVED** [review]
Feature has been tested and satisfies required criteria.

**Pull request merged**